File: src/main/java/com/dh/ondot/member/api/response/PreparationTimeResponse.java
package com.dh.ondot.member.api.response;

import com.dh.ondot.core.util.TimeUtils;
import com.dh.ondot.member.domain.Member;

import java.time.LocalDateTime;

public record PreparationTimeResponse(
        Integer preparationTime,
        LocalDateTime updatedAt
) {
    public static PreparationTimeResponse from(Member member) {
        return new PreparationTimeResponse(
                member.getPreparationTime(),
                TimeUtils.toSeoulDateTime(member.getUpdatedAt())
        );
    }
}

File: src/main/kotlin/com/dh/ondot/member/api/response/PreparationTimeResponse.kt
package com.dh.ondot.member.api.response

import com.dh.ondot.core.util.TimeUtils
import com.dh.ondot.member.domain.Member
import java.time.LocalDateTime

data class PreparationTimeResponse(
    val preparationTime: Int,
    val updatedAt: LocalDateTime,
) {
    companion object {
        fun from(member: Member): PreparationTimeResponse =
            PreparationTimeResponse(
                preparationTime = member.preparationTime!!,
                updatedAt = TimeUtils.toSeoulDateTime(member.updatedAt)!!,
            )
    }
}
